text/0000-serve-ros-map-from-pbstream.md

```diff
@@ -0,0 +1,26 @@
+# Serve ROS map from a pbstream file
+
+## Summary
+[summary]: #summary
+
+We would like to have a ROS node that serves a `nav_msgs/OccupancyGrid` message like the `occupancy_grid_node` does.
+But instead of publishing the current state of cartographer, a frozen state from a pbstream file should be read and compiled into a flat map, which is then published.
+
+
+## Motivation
+[motivation]: #motivation
+
+Since `OccupancyGrid` is the standard message type for map, many nodes depend on such a topic.
+If the pbstream gets frequently updated re-generating a `.pgm` and `.yaml` file in oder to read it in again using a ROS `map_server` is administrative overhead and generates redundant data. 
```

Comment:
If the pbstream is often updated, how would this be solved if you had this binary which publishes the map as a topic? Would this binary be watching the file? Would you need to restart the binary?

Reply:
Good question. We were actually just thinking of a static map server that reads the pbstream file once. But maybe a watch_and_update command line flag would be worth thinking about. 
Do you have a use case in mind that is not already covered by the occupancy grid node?